### PR TITLE
finished project for  W2 Ceylan Baysal

### PR DESCRIPTION
**Pull request by ceylanbaysalaydin**

### CODE REVIEW

File: Week2/js-exercises/collectiveAges.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    { name: 'Tjebbe', age: 22 },
+  ];
+  const add = (a, b) => a + b;
+  function collectiveAges(array, callback) {
+    const ages = array.map(item => item.age);
+    const collectiveAge = ages.reduce(callback);
+    console.log(`The collective age of the HYF team is: ${collectiveAge}`);
+    return collectiveAge;
+  }
+  const result = collectiveAges(hackYourFutureMembers, add);
```
Comment (wimklerkx) [inline Week2/js-exercises/collectiveAges.js]: Good code: working well and readable!

File: Week2/js-exercises/monday.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      name: 'Development time',
+      duration: 240,
+    },
+    {
+      name: 'Talk to different members from the product team',
+      duration: 60,
+    },
+  ];
+  const hourlyRate = 25;
+  function calculateMoney(taskObject, hourRate) {
```
Comment (wimklerkx) [inline Week2/js-exercises/monday.js]: What would be a better, more precise name for this function?

File: Week2/js-exercises/monday.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    let totalDuration = 0;
+    let totalHour = 0;
+    const euro = '€';
+    taskObject.map(item => (totalDuration += item.duration));
+    totalHour = totalDuration / 60;
+    const earnedMoney = (totalHour * hourRate).toFixed(2);
+    return euro + earnedMoney;
+  }
```
Comment (wimklerkx) [inline Week2/js-exercises/monday.js]: Good and readable code, well done!

File: Week2/js-exercises/lemonAllergy.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+{
+  const fruitBasket = ['Apple', 'Lemon', 'Grapefruit', 'Lemon', 'Banana', 'Watermelon', 'Lemon'];
+  const takeOutLemon = array => {
+    const outLemon = array.filter(item => item !== 'Lemon');
```
Comment (wimklerkx) [inline Week2/js-exercises/lemonAllergy.js]: What would be a better name for the variable holding the basket without Lemons?

File: Week2/js-exercises/lemonAllergy.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+    return `My mom bought me a fruit basket, containing ${outLemon.join(', ')}!`;
+  };
+  console.log(takeOutLemon(fruitBasket));
```
Comment (wimklerkx) [inline Week2/js-exercises/lemonAllergy.js]: Well used filter function!

File: Week2/js-exercises/oddOnesOut.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+  // Using the map and filter functions, rewrite the doubleEvenNumbers function.
+}
+{
+  function doubleEvenNumbers(numbers) {
+    const evenNumbers = numbers.filter(item => item % 2 === 0);
+    const doubleEven = evenNumbers.map(item => item * 2);
+    return doubleEven;
+  }
+  const myNumbers = [1, 2, 3, 4];
+  console.log(doubleEvenNumbers(myNumbers));
```
Comment (wimklerkx) [inline Week2/js-exercises/oddOnesOut.js]: Wonderfully clear and concise. And it works!

File: Week2/js-exercises/hobbies.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+    'Reading',
+    'Programming',
+    'Hanging out with friends',
+    'Going to the gym',
+  ];
+  function createAndAppend(create, appendTo) {
+    const element = document.createElement(create);
+    appendTo.appendChild(element);
+    return element;
+  }
```
Comment (wimklerkx) [inline Week2/js-exercises/hobbies.js]: Nice and re-usable function

File: Week2/js-exercises/hobbies.js [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+  const h1 = createAndAppend('h1', document.body);
+  h1.innerHTML = 'My Hobbies';
+  const ul = createAndAppend('ul', document.body);
+  myHobbies.forEach(item => {
+    const li = createAndAppend('li', ul);
+    li.innerHTML = item;
+  });
```
Comment (wimklerkx) [inline Week2/js-exercises/hobbies.js]: That's all it needs :)

File: Week2/project/script.js [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+        setTimeout(playSeconds, 1000);
+        running = false;
+      }
+      if (!minutes && !seconds) {
+        // When timer is 00:00 calling the timesUp function
+        timesUp();
+      }
+    }, 1000);
+  }
+}
```
Comment (wimklerkx) [inline Week2/project/script.js]: Generally it works very well.

The only problem I can find is when clicking play or pause multiple times. The clock seems to go faster each time I click play. What might be the problem?

Comment (ceylanbaysalaydin) [reply Week2/project/script.js]: You are right @wimklerkx . I can make an if statement at the top of the playSeconds function an everything inside of the function will be inside of that if. Like if( running === false){...}
So if time is already running playSecond function will be called after click but do nothing.I guess this can be solve the problem. I have never tried clicking multiple times any of them. I will change it as soon as possible. I am learning also how to test my projects thanks to you.🙂💐

File: Week2/js-exercises/lemonAllergy.js (hunk 4 above)
Comment (ceylanbaysalaydin) [reply Week2/js-exercises/lemonAllergy.js]: Maybe, basketWithoutLemon would be better 😅
I will think twice while naming things, thanks a lot @wimklerkx

**Comment by ceylanbaysalaydin** [comment]
> You write good and readable code. And you understand the challenging topics of this week. With a little more attention to naming things and to testing edge cases and user-abuse, you could improve even more!

Thanks a lot  for your valuable feedbacks @wimklerkx . I will keep in mind all of them. I appreciate your help. 🌹

**Comment by wouterkleijn** [comment]
Well done @ceylanbaysalaydin. As I have said before: I like how you respond to Wim's comments as well - very professional :)

**Comment by ceylanbaysalaydin** [comment]
> Well done @ceylanbaysalaydin. As I have said before: I like how you respond to Wim's comments as well - very professional :)

Thank you so much @wouterkleijn 💐
